internal/pkg/metrics/alerts/alerts.go

```diff
@@ -0,0 +1,198 @@
+	return &monitoringv1.Rule{
+		Alert: string(flowslatest.AlertNoFlows),
+		Annotations: map[string]string{
+			"description": "NetObserv flowlogs-pipeline is not receiving any flow, this is either a connection issue with the agent, or an agent issue",
+			"summary":     "NetObserv flowlogs-pipeline is not receiving any flow",
+		},
+		Expr:   intstr.FromString("sum(rate(netobserv_ingest_flows_processed[1m])) == 0"),
+		For:    &d,
+		Labels: buildLabels("warning", false),
+	}
+}
+
+func alertLokiError() *monitoringv1.Rule {
+	d := monitoringv1.Duration("10m")
+
+	return &monitoringv1.Rule{
+		Alert: string(flowslatest.AlertLokiError),
+		Annotations: map[string]string{
+			"description": "NetObserv flowlogs-pipeline is dropping flows because of Loki errors, Loki may be down or having issues ingesting every flows. Please check Loki and flowlogs-pipeline logs.",
+			"summary":     "NetObserv flowlogs-pipeline is dropping flows because of Loki errors",
+		},
+		Expr:   intstr.FromString("sum(rate(netobserv_loki_dropped_entries_total[1m])) > 0"),
+		For:    &d,
+		Labels: buildLabels("warning", false),
+	}
+}
+
+func (rb *ruleBuilder) kernelDrops() (*monitoringv1.Rule, error) {
+	description := fmt.Sprintf(
+		"NetObserv is detecting more than %s%% of packets dropped by the kernel%s. %s",
+		rb.threshold,
+		rb.getAlertLegend(),
+		rb.additionalDescription(),
+	)
+
+	metric, totalMetric := rb.getMetricsForAlert()
+	metricsRate := promQLRateFromMetric(metric, "", "", "2m", "")
+	totalRate := promQLRateFromMetric(totalMetric, "", "", "2m", "")
+	metricsSumBy := sumBy(metricsRate, rb.alert.GroupBy, rb.side, "")
+	totalSumBy := sumBy(totalRate, rb.alert.GroupBy, rb.side, "")
+	promql := percentagePromQL(metricsSumBy, totalSumBy, rb.threshold, rb.upperThreshold, rb.alert.LowVolumeThreshold)
+
+	return rb.createRule(promql, "Too many packet drops by the kernel", description)
+}
+
+func (rb *ruleBuilder) deviceDrops() (*monitoringv1.Rule, error) {
+	// No "side" consideration on netdev metrics, so keep only 1 call from the two of them
+	if rb.side == asDest {
+		return nil, nil
+	}
+	var byLabels string
+	var healthAnnotOverride map[string]any
+	var legend string
+	switch rb.alert.GroupBy {
+	case flowslatest.GroupByNode:
+		byLabels = " by (instance)"
+		healthAnnotOverride = map[string]any{"nodeLabels": "instance"}
+		legend = " [node={{ $labels.instance }}]"
+	case flowslatest.GroupByNamespace:
+		return nil, fmt.Errorf("PacketDropsByDevice alert does not support grouping per namespace")
+	case flowslatest.GroupByWorkload:
+		return nil, fmt.Errorf("PacketDropsByDevice alert does not support grouping per workload")
+	}
+
+	promql := percentagePromQL(
+		fmt.Sprintf("sum(rate(node_network_receive_drop_total[2m]))%s + sum(rate(node_network_transmit_drop_total[2m]))%s", byLabels, byLabels),
+		fmt.Sprintf("sum(rate(node_network_receive_packets_total[2m]))%s + sum(rate(node_network_transmit_packets_total[2m]))%s", byLabels, byLabels),
```

Comment:
Should we expose the range (2m) here ? I guess it can impact a lot on the thresholds you are setting in some cases.

Reply:
I'm fine with that if we are feeling it's enough for now as we don't cover much cases yet